sycl/test-e2e/KernelFusion/internalize_array_wrapper.cpp

```diff
@@ -2,6 +2,9 @@
 // RUN: %{build} -fsycl-embed-ir -O2 -o %t.out
 // RUN: %{run} %t.out
 
+// FIXME: enable opaque pointers support
+// REQUIRES: TEMPORARY_DISABLED
+
```

Comment:
FYI: compiler gives me a warning that -fsycl-embed-ir is ignored.
Kernel fusion developers might need to check if this option is needed for test.

Reply:
The option is only needed on non-SPIR-V targets (currently only CUDA), so it's fine if the option is ignored when compiling for SPIR-V. 

I assume the test only failed when running it, not during compilation, right?

Reply:
> I assume the test only failed when running it, not during compilation, right?

@sommerlukas here is the error message from the [pre-commit logs](https://github.com/intel/llvm/actions/runs/5731980488/job/15535109389).

```
FAIL: SYCL :: KernelFusion/internalize_array_wrapper.cpp (1038 of 1567)
******************** TEST 'SYCL :: KernelFusion/internalize_array_wrapper.cpp' FAILED ********************
Script:
--
: 'RUN: at line 2';    /__w/llvm/llvm/toolchain/bin//clang++   -fsycl -fsycl-targets=spir64 /__w/llvm/llvm/llvm/sycl/test-e2e/KernelFusion/internalize_array_wrapper.cpp -Xclang -no-opaque-pointers -fsycl-embed-ir -O2 -o /__w/llvm/llvm/build-e2e/KernelFusion/Output/internalize_array_wrapper.cpp.tmp.out
: 'RUN: at line 3';   env ONEAPI_DEVICE_SELECTOR=ext_oneapi_level_zero:gpu  /__w/llvm/llvm/build-e2e/KernelFusion/Output/internalize_array_wrapper.cpp.tmp.out
: 'RUN: at line 3';   env ONEAPI_DEVICE_SELECTOR=opencl:gpu  /__w/llvm/llvm/build-e2e/KernelFusion/Output/internalize_array_wrapper.cpp.tmp.out
: 'RUN: at line 3';   env ONEAPI_DEVICE_SELECTOR=opencl:cpu  /__w/llvm/llvm/build-e2e/KernelFusion/Output/internalize_array_wrapper.cpp.tmp.out
--
Exit Code: 1

Command Output (stdout):
--
$ ":" "RUN: at line 2"
note: command had no output on stdout or stderr
$ "/__w/llvm/llvm/toolchain/bin//clang++" "-fsycl" "-fsycl-targets=spir64" "/__w/llvm/llvm/llvm/sycl/test-e2e/KernelFusion/internalize_array_wrapper.cpp" "-Xclang" "-no-opaque-pointers" "-fsycl-embed-ir" "-O2" "-o" "/__w/llvm/llvm/build-e2e/KernelFusion/Output/internalize_array_wrapper.cpp.tmp.out"
# command stderr:
clang++: warning: argument unused during compilation: '-fsycl-embed-ir' [-Wunused-command-line-argument]
/__w/llvm/llvm/toolchain/bin/llvm-link: /tmp/lit-tmp-fznnrqd4/libsycl-crt-8cab33.o: error: Opaque pointers are only supported in -opaque-pointers mode (Producer: 'LLVM17.0.0git' Reader: 'LLVM 17.0.0git')
/__w/llvm/llvm/toolchain/bin/llvm-link: error:  loading file '/tmp/lit-tmp-fznnrqd4/libsycl-crt-8cab33.o'
clang++: error: sycl-link command failed with exit code 1 (use -v to see invocation)

error: command failed with exit status: 1

--
```

It looks like JIT compiler for fusion component still uses typed pointers. Am I right?

Reply:
In that case I would expect most kernel fusion related tests to fail, not just this one. 

Also, the error is from `sycl-link`, so already seems to happen at compile time, whereas errors in the JIT compiler would only show up at runtime of the application. 

Is the problem maybe that `"-Xclang" "-no-opaque-pointers" ` is still passed in the compilation of the test?

Reply:
> Is the problem maybe that "-Xclang" "-no-opaque-pointers" is still passed in the compilation of the test?

I don't see these options in the command line. Why do you think they are still passed?

Reply:
Because of this line from the error log you posted: 
```
"/__w/llvm/llvm/toolchain/bin//clang++" "-fsycl" "-fsycl-targets=spir64" "/__w/llvm/llvm/llvm/sycl/test-e2e/KernelFusion/internalize_array_wrapper.cpp" "-Xclang" "-no-opaque-pointers" "-fsycl-embed-ir" "-O2" "-o" "/__w/llvm/llvm/build-e2e/KernelFusion/Output/internalize_array_wrapper.cpp.tmp.out"
```

Reply:
Which corresponds to the first `RUN` line in the log.

Reply:
Right. This was my attempt to fix [this error](https://github.com/intel/llvm/actions/runs/5730435105/job/15532799242):

```
FAIL: SYCL :: KernelFusion/internalize_array_wrapper.cpp (1041 of 1567)
******************** TEST 'SYCL :: KernelFusion/internalize_array_wrapper.cpp' FAILED ********************
Script:
--
: 'RUN: at line 2';    /__w/llvm/llvm/toolchain/bin//clang++   -fsycl -fsycl-targets=spir64 /__w/llvm/llvm/llvm/sycl/test-e2e/KernelFusion/internalize_array_wrapper.cpp -fsycl-embed-ir -O2 -o /__w/llvm/llvm/build-e2e/KernelFusion/Output/internalize_array_wrapper.cpp.tmp.out
: 'RUN: at line 3';   env ONEAPI_DEVICE_SELECTOR=ext_oneapi_level_zero:gpu  /__w/llvm/llvm/build-e2e/KernelFusion/Output/internalize_array_wrapper.cpp.tmp.out
: 'RUN: at line 3';   env ONEAPI_DEVICE_SELECTOR=opencl:gpu  /__w/llvm/llvm/build-e2e/KernelFusion/Output/internalize_array_wrapper.cpp.tmp.out
: 'RUN: at line 3';   env ONEAPI_DEVICE_SELECTOR=opencl:cpu  /__w/llvm/llvm/build-e2e/KernelFusion/Output/internalize_array_wrapper.cpp.tmp.out
--
Exit Code: -6

Command Output (stdout):
--
$ ":" "RUN: at line 2"
note: command had no output on stdout or stderr
$ "/__w/llvm/llvm/toolchain/bin//clang++" "-fsycl" "-fsycl-targets=spir64" "/__w/llvm/llvm/llvm/sycl/test-e2e/KernelFusion/internalize_array_wrapper.cpp" "-fsycl-embed-ir" "-O2" "-o" "/__w/llvm/llvm/build-e2e/KernelFusion/Output/internalize_array_wrapper.cpp.tmp.out"
# command stderr:
clang++: warning: argument unused during compilation: '-fsycl-embed-ir' [-Wunused-command-line-argument]

$ ":" "RUN: at line 3"
note: command had no output on stdout or stderr
$ "env" "ONEAPI_DEVICE_SELECTOR=ext_oneapi_level_zero:gpu" "/__w/llvm/llvm/build-e2e/KernelFusion/Output/internalize_array_wrapper.cpp.tmp.out"
# command stderr:
internalize_array_wrapper.cpp.tmp.out: /__w/llvm/llvm/llvm/sycl/test-e2e/KernelFusion/internalize_array_wrapper.cpp:131: int main(): Assertion `all(tmp[id][i][j] == not_written) && "Not internalizing"' failed.

error: command failed with exit status: -6

--
```

Reply:
Oh, sorry, I misunderstood your previous comment, my bad. I missed that adding the non-opaque flag was your attempt to fix the test. 

I'll approve the changes to the test to deactivate it now.